sos/plugins/kubernetes.py

```diff
@@ -22,33 +23,101 @@ class kubernetes(Plugin, RedHatPlugin):
     """Kubernetes plugin
     """
 
-    option_list = [("podslog", "capture logs for pods", 'slow', False)]
+    # Red Hat Atomic Platform and OpenShift Enterprise use the
+    # atomic-openshift-master package to provide kubernetes
+    packages = ('kubernetes', 'atomic-openshift-master')
```

Comment:
Right, indeed it's that 'extrospection' use case that I was talking about when I mentioned "a containerized install" of a kubernetes master. Yes, the host does have packages and you can query them, but here we'd be checking for packages that are installed inside a container.

Supporting this use case in general for sos is beyond the scope of this PR.

For this plugin in particular though the situation is hopefully simpler while we wait for that extra support in core sos: even if the kube master is running as a container the plugin does not really need to introspect much of the container itself: it basically queries the API of the kube master (running on a container or not) whose URL is stored in a config file outside of the container (same location as a non-container install). It does rely on the client tool (`kubectl`) to do that though, and it would fail if that's not available where sos runs. I believe that implementing those API calls directly in the plugin without relying on `kubectl` is also beyond scope.

However, `kubectl` is one of the few tools that are shipped in base Atomic Host (and of course it should also be available when the installation is direclty on the host).

We don't want kubernetes client tools as a dependency for sos, but I believe it's reasonable to assume that these will be available in the common scenarios where the plugin would be used, at least for the foreseeable future (certainly my view of the future is quite narrow and short though). If `kubectl` is not available the plugin would fail... is this a bigger problem than the plugin not auto-enabling on this common deployment scenario?